integration/qbft/scenarios/regular.go

```diff
@@ -57,115 +61,160 @@ func Regular(role spectypes.BeaconRole) *IntegrationTest {
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       proposalData,
-					}),
-				},
-			}},
-			PrepareContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{
-				specqbft.FirstRound: {
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[1], 1, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[2], 2, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[3], 3, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-					spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[4], 4, &specqbft.Message{
-						MsgType:    specqbft.PrepareMsgType,
-						Height:     specqbft.FirstHeight,
-						Round:      specqbft.FirstRound,
-						Identifier: identifier[:],
-						Data:       prepareData,
-					}),
-				},
-			}},
-			// NOTE: need to keep round keys and commit message data, there's a special check for CommitContainer
-			CommitContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{
-				specqbft.FirstRound: {
-					&specqbft.SignedMessage{
-						Message: &specqbft.Message{
-							Data: commitData,
-						},
-					},
-				},
-			}},
-			RoundChangeContainer: &specqbft.MsgContainer{Msgs: map[specqbft.Round][]*specqbft.SignedMessage{}},
-		},
-		DecidedMessage: &specqbft.SignedMessage{
-			Message: &specqbft.Message{
+				Data:       prepareData,
+			})
+			if !isMessageExistInRound(expectedMsg, actual.State.PrepareContainer.Msgs[specqbft.FirstRound]) {
+				return fmt.Errorf("prepare message %+v wasn't found at actual.State.PrepareContainer.Msgs[specqbft.FirstRound]", expectedMsg)
+			}
+
+			expectedMsg = spectestingutils.SignQBFTMsg(spectestingutils.Testing4SharesSet().Shares[operatorIDIterator], operatorIDIterator, &specqbft.Message{
 				MsgType:    specqbft.CommitMsgType,
 				Height:     specqbft.FirstHeight,
 				Round:      specqbft.FirstRound,
 				Identifier: identifier[:],
-				Data:       spectestingutils.PrepareDataBytes(consensusData),
+				Data:       commitData,
+			})
+			if isMessageExistInRound(expectedMsg, actual.State.CommitContainer.Msgs[specqbft.FirstRound]) {
+				foundedMsgsCounter++
+			}
+		}
+
+		if !actual.State.Share.HasQuorum(foundedMsgsCounter) {
```

Comment:
@egorka-sereda 

If there're two messages from 1 and one message from 2, it will pass but should fail. We need to make sure that:
- each actual message exists in a set of possible messages
- it is not duplicated
- there are 3 actual messages (I think we won't have a case when we have 4 actual messages, let's check if it's 3)